pkg/odo/cli/component/create.go

```diff
@@ -412,18 +411,6 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 		co.devfileMetadata.componentName = strings.ToLower(componentName)
 		co.devfileMetadata.componentNamespace = strings.ToLower(componentNamespace)
 
-		// If devfile.yaml is present, we don't need to download the devfile.yaml later
-		if util.CheckPathExists(DevfilePath) {
```

Comment:
this check is not needed because we are already checking it above.

Reply:
This is needed because:

I don't think we can do the global check for devfile existence here, and we have to check separately in interactive mode and direct mode, reason is:
- Interactive mode: we can use devfile directly if devfile is present rather than error out, use can still specify the component name via interactive mode
- Direct mode: we need to error out as component type that user specify may have conflict with the existing devfile

Reply:
done